dcos_launch/gce.py

```diff
@@ -0,0 +1,53 @@
+import json
+import logging
+
+from dcos_launch import util
+from dcos_test_utils import gce
+from dcos_test_utils.helpers import Host
+
+log = logging.getLogger(__name__)
+
+
+class BareClusterLauncher(util.AbstractLauncher):
+    # Launches a homogeneous cluster of plain GMIs intended for onprem DC/OS
+    def __init__(self, config):
+        credentials_path = util.set_from_env('GOOGLE_APPLICATION_CREDENTIALS')
+
+        with open(credentials_path, 'r') as creds:
+            self.gce_wrapper = gce.GceWrapper(json.load(creds))
+
+        self.config = config
+        self.deployment = gce.Deployment(self.gce_wrapper, config['deployment_name'], config['zone'])
+        self.insert_request_errors = None
+
+    def create(self) -> dict:
+        response = self.gce_wrapper.deploy_instances(self.config)
+        self.insert_request_errors = response.get('error')
+        self.key_helper()
+        return self.config
+
+    def key_helper(self):
+        # Generate the private key file and applies the public key to the instance group
+        if not self.config.get('key_helper'):
```

Comment:
Should just do config['key_helper'] unless its a situation where we expect the key to not possibly exist